jsapp/js/components/submissions/submissionModal.es6

```diff
@@ -476,6 +486,16 @@ class SubmissionModal extends React.Component {
                 </a>
               }
 
+              {this.userCan('view_submissions', this.props.asset) &&
+                <a
+                  onClick={this.launchViewSubmission.bind(this)}
+                  className='kobo-button kobo-button--blue submission-duplicate__button'
```

Comment:
I think when you click it, it should be disabled while loading, so `disabled={this.state.isViewLoading}` :)